hack/kind/setup.sh

```diff
@@ -0,0 +1,80 @@
+KIND_IMAGE_K8S_133=docker.io/kindest/node:v1.33.4@sha256:25a6018e48dfcaee478f4a59af81157a437f15e6e140bf103f85a2e7cd0cbbf2
+KIND_IMAGE_K8S_134=docker.io/kindest/node:v1.34.0@sha256:7416a61b42b1662ca6ca89f02028ac133a309a2a30ba309614e8ec94d976dc5a
+
+kind_cluster_name=${name}
+
+if printenv K8S_VERSION >/dev/null && [ -n "$K8S_VERSION" ]; then
+  k8s_version="$K8S_VERSION"
+fi
+
+case "$k8s_version" in
+1.31*) image=$KIND_IMAGE_K8S_131 ;;
+1.32*) image=$KIND_IMAGE_K8S_132 ;;
+1.33*) image=$KIND_IMAGE_K8S_133 ;;
+1.34*) image=$KIND_IMAGE_K8S_134 ;;
+v*) printf "${red}${redcross}Error${end}: Kubernetes version must be given without the leading 'v'\n" >&2 && exit 1 ;;
+*) printf "${red}${redcross}Error${end}: unsupported Kubernetes version ${yel}${k8s_version}${end}\n" >&2 && exit 1 ;;
+esac
+
+echo "$image"
+
+setup_kind() {
+  # (0) If kind is not installed, install it
+  if ! command -v kind >/dev/null 2>&1; then
+    printf "${red}${redcross}Error${end}: kind is not installed. Please install kind: https://kind.sigs.k8s.io/docs/user/quick-start/#installation${end}\n" >&2
+    exit 1
+  fi
+
+  # (1) Does the kind cluster already exist?
+  if ! kind get clusters -q | grep -q "^$kind_cluster_name\$"; then
+    kind create cluster --config "./hack/kind/cluster.yaml" \
+      --image "$image" \
+      --name "$kind_cluster_name"
+  fi
+
+  # (2) Does the kube config contain the context for this existing kind cluster?
+  if ! kubectl config get-contexts -oname 2>/dev/null | grep -q "^kind-${kind_cluster_name}$"; then
+    printf "${red}${redcross}Error${end}: the kind cluster ${yel}$kind_cluster_name${end} already exists, but your current kube config does not contain the context ${yel}kind-$kind_cluster_name${end}. Run:\n" >&2
+    printf "    ${cyan}kind delete cluster --name $kind_cluster_name${end}\n" >&2
+    printf "and then retry.\n"
+    exit 1
+  fi
+
+  # (3) Is the existing kind cluster selected as the current context in the kube
+  # config?
+  if [ "$(kubectl config current-context 2>/dev/null)" != "kind-$kind_cluster_name" ]; then
+    printf "${red}${redcross}Error${end}: the kind cluster ${yel}$kind_cluster_name${end} already exists, but is not selected as your current context. Run:\n" >&2
+    printf "    ${cyan}kubectl config use-context kind-$kind_cluster_name${end}\n" >&2
+    exit 1
+  fi
+
+  # (4) Is the current context responding?
+  if ! kubectl --context "kind-$kind_cluster_name" get nodes >/dev/null 2>&1; then
+    printf "${red}${redcross}Error${end}: the kind cluster $kind_cluster_name isn't responding. Please run:\n" >&2
+    printf "    ${cyan}kind delete cluster --name $kind_cluster_name${end}\n" >&2
+    printf "and then retry.\n"
+    exit 1
+  fi
+
+  # (5) Does the current context have the correct Kubernetes version?
+  existing_version=$(kubectl version -oyaml | yq e '.serverVersion | .major +"."+ .minor' -)
```

Comment:
Be aware, we don't use yq anywhere else in the repo. We would need to install yq just for this command in the CI then

Reply:
yes it's good catch. I'll fix it.

Reply:
it's very handy - i used it daily :)

Reply:
@albertogdd @gkrenn done in 2f955a076